app/controllers/application_controller.rb

```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  def require_user
+    redirect_to sign_in_path unless current_user
+  end
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```

Comment:
Thanks fixed.  I actually thought about memoisation and then forgot to implement.  However, I do note that this isn't included into the solution video which made me think maybe it wasn't necessary.